src/components/AsideHeader/components/AllPagesPanel/AllPagesPanel.tsx

```diff
@@ -47,6 +46,21 @@ export const AllPagesPanel: React.FC<AllPagesPanelProps> = (props) => {
 
     const [isEditMode, setIsEditMode] = useState(false);
 
+    const [collapsedGroups, setCollapsedGroups] = useState<Record<string, boolean>>(() => {
+        const initial: Record<string, boolean> = {};
+
+        items.forEach((item) => {
+            if ('groupId' in item && item.groupId && 'isCollapsed' in item) {
+                initial[item.groupId] = Boolean(true);
+            }
+        });
+        return initial;
+    });
```

Comment:
And if items change, don't you need to recalculate the state?

Reply:
No, because I have separated this state from props items. It's should be independent.